src/app/containers/ConsentBanner/CanonicalLogic/index.js

```diff
@@ -42,14 +52,12 @@ const setSeenPrivacyBanner = () =>
     value: PRIVACY_COOKIE_CURRENT,
     sameSite: SAME_SITE_VALUE,
   });
-const setDefaultPolicy = logger => setPolicyCookie(POLICY_DENIED, logger);
-const setApprovedPolicy = logger => setPolicyCookie(POLICY_APPROVED, logger);
-const setDismissedCookieBanner = () =>
-  setCookie({
-    name: EXPLICIT_COOKIE,
-    value: COOKIE_BANNER_APPROVED,
-    sameSite: SAME_SITE_VALUE,
-  });
+const setDefaultPolicy = logger =>
+  setPolicyCookie(POLICY_DENIED, logger, false);
+const setApprovedPolicy = logger =>
+  setPolicyCookie(POLICY_APPROVED, logger, true);
+const setDismissedCookieBanner = logger =>
+  setPolicyCookie(POLICY_DENIED, logger, true);
```

Comment:
This seems to contradict the old code (line 50): `value: COOKIE_BANNER_APPROVED,`

Did we have a bug previously, the new code seems more correct?
For the rejection path, we just set denied policy: https://github.com/bbc/simorgh/pull/9271/files#diff-5b7ad8946ed3ee415f62ce965a3f465e2e3152e45934793b7600767cc4d3eb93R108
For the accept path we set the denied policy then immediately follow by setting the approved policy:
https://github.com/bbc/simorgh/pull/9271/files#diff-5b7ad8946ed3ee415f62ce965a3f465e2e3152e45934793b7600767cc4d3eb93R102-R103

sorry if I'm getting confused?

Reply:
Oh, you're right. I think I made assumptions without reading it properly.

I'll say what I intended and we can see if I've made a mess of it or not:

- I believe `setDefaultPolicy` is what's called when the banner is shown before the user does anything. That needs to set `000` and with the `explicit` param set to `false` so that it only does so via JS and on the current domain. No ovens involved.
- I understood `setApprovedPolicy` to be when people say "I accept" and click it, so we set to `111` and `explicit` is `true` because we want to set via cookie oven call instead of JS and we can do so on both dmains.
- Then I thought `setDismissedCookieBanner` was when people click to reject or something so I went for `000` and to set it via HTTP and cross-domain. Looking at it now, that's now what it's doing it is?

I think on closer reading, this is called now just straight after, say, `setApprovedPolicy` to additionally set the explicit cookie and stop the banner sticking. Since calling `setApprovedPolicy` in this PR will hit the new oven URL which will set that for you, that would mean `setDismissedCookieBanner` is entirely redundant?